datastore/warehouse/structs.go

```diff
@@ -0,0 +1,53 @@
+package warehouse
+
+import (
+	"errors"
+	"os"
+	"time"
+)
+
+var (
+	ErrUnknownType = errors.New("unknown data type")
+)
+
+type DataType uint8
+
+const (
+	GetPayloadRequest DataType = iota
+	SubmitBlockRequest
+)
+
+func toString(data DataType) string {
+	switch data {
+	case GetPayloadRequest:
+		return "GetPayloadRequest"
+	case SubmitBlockRequest:
+		return "SubmitBlockRequest"
+	default:
+		return "unknown"
+	}
+}
+
+type StoreRequest struct {
+	DataType  DataType
```

Comment:
Assuming that we would like to introduce that in more places - do we need this to be defined defined? 
You're not really switching on that, it's only used for logging. `DataType string` would be more than enough. and as we pass A lot already, few more is not tragedy

Reply:
Very good idea!